dbms/src/Storages/DeltaMerge/File/DMFile.h

```diff
@@ -303,14 +303,18 @@ class DMFile : private boost::noncopyable
      * Note that only the column id and type is valid.
      * @return All columns
      */
-    ColumnDefines getColumnDefines()
+    ColumnDefines getColumnDefines(bool sort_by_id = true)
     {
         ColumnDefines results{};
         results.reserve(this->column_stats.size());
         for (const auto & cs : this->column_stats)
         {
             results.emplace_back(cs.first, "", cs.second.type);
         }
+        if (sort_by_id)
+            std::sort(results.begin(), results.end(), [](const auto & lhs, const auto & rhs) {
```

Comment:
Is it better to use cbegin and cend?

Reply:
no, sort will change the vector, can not use const iterator.